src/main/java/io/stargate/sgv2/jsonapi/api/model/command/builders/TableSortClauseBuilder.java

```diff
@@ -31,10 +33,53 @@ public TableSortClauseBuilder(TableSchemaObject table) {
   @Override
   protected SortClause buildClauseFromDefinition(ObjectNode sortNode) {
     // First, resolve the paths to column definitions
-    List<SortExpressionDefinition> sortExprDefs = resolveColumns(sortNode);
-    final List<SortExpression> sortExpressions = new ArrayList<>();
+    var sortExprDefs = resolveColumns(sortNode);
+
+    // Then split into "special" (vector/vectorize, lexical) and regular expressions
+    var lexicalExprs = new ArrayList<SortExpressionDefinition>();
+    var vectorExprs = new ArrayList<SortExpressionDefinition>();
+    var regularExprs = new ArrayList<SortExpressionDefinition>();
+
     for (SortExpressionDefinition sortExprDef : sortExprDefs) {
-      sortExpressions.add(buildSortExpression(sortExprDef));
+      var column = sortExprDef.column;
+      switch (column.type().typeName()) {
+        case VECTOR -> vectorExprs.add(sortExprDef);
+        case ASCII, TEXT -> {
+          if (sortExprDef.sortValue.isTextual()) {
+            lexicalExprs.add(sortExprDef);
+          } else {
+            regularExprs.add(sortExprDef);
+          }
+        }
+        default -> regularExprs.add(sortExprDef);
+      }
+    }
+
+    // Lexical(s)? Must have but one expression, cannot be combined with other sorts
+    // Ditto for vector/vectorize
+    if (!lexicalExprs.isEmpty() || !vectorExprs.isEmpty()) {
+      if (sortExprDefs.size() > 1) {
+        throw SortException.Code.CANNOT_SORT_ON_SPECIAL_WITH_OTHERS.get(
+            errVars(
+                schema,
+                map -> {
+                  map.put("lexicalSorts", columnsDesc(lexicalExprs));
+                  map.put("regularSorts", columnsDesc(regularExprs));
+                  map.put("vectorSorts", columnsDesc(vectorExprs));
+                }));
+      }
+
+      if (!lexicalExprs.isEmpty()) {
```

Comment:
After basic validation on Column type, constructing expression: actual validation of Index (and other limits) handled later on in the processing pipeline.
(formerly was not validating Column type here at all).